File: Back/src/DotEco.Domain/CollectionData.cs
using System;
using DotEco.Domain.Identity;
using System.ComponentModel.DataAnnotations.Schema;

namespace DotEco.Domain
{
    [Table("CollectionData")]
    public class CollectionData
    {
        [Column("CollectionDataId")]
        public int Id { get; set; }
        public string Name { get; set; }
        public string Address { get; set; }
        public string CEP { get; set; }
        public string Reference { get; set; }
        public string Email { get; set; }
        public string Telephone { get; set; }
        public string TypeCollection { get; set; }
        public DateTime Date { get; set; }
        public int AssociationId { get; set; }
        public int? UserId { get; set; }
        public User User { get; set; }
        public StatusClient StatusClient { get; set; }
        public StatusAssociation StatusAssociation { get; set; }
    }
}

public enum StatusClient
{
    Waiting = 0,
    Scheduled = 1,
    Confirmed = 2,
    Refused = 3,
}

public enum StatusAssociation
{
    Waiting = 0,
    Scheduled = 1,
    Confirmed = 2,
    Refused = 3,
}

File: Front/DotEco-APP/src/app/_models/CollectionData.ts
import { Association } from "./Association";

export interface CollectionData {
    id: number;
    name: string;
    address: string;
    cep: string;
    reference: string;
    email: string;
    telephone: string;
    userId: number;
    associationId: Association[];
    statusPoint: StatusPoint;
    statusClient: StatusClient;
    statusAssociation: StatusAssociation;
}

export enum StatusClient {
    Waiting = "Aguardando",
    Scheduled = "Agendado",
    Confirmed = "Confirmado",
    Refused = "Recusado",
}

export enum StatusAssociation {
    Waiting = "Aguardando",
    Scheduled = "Agendado",
    Confirmed = "Confirmado",
    Refused = "Recusado",
}

export enum StatusPoint {
    NotRescued = "NaoResgatado",
    Rescued = "Resgatado",
}